eth_tester/backends/pyevm/main.py

```diff
@@ -278,7 +288,7 @@ class PyEVMBackend(object):
     chain = None
     fork_config = None
 
-    def __init__(self):
+    def __init__(self, genesis_parameter_overrides=None):
```

Comment:
I think it would be better to just require providing the fully merged genesis parameters.  I believe that will simplify a lot of these apis, removing the concept of merging the overrides into the base parameters.  We should however provide an API for generating the merged parameters to make this easy for users.

Maybe a `classmethod` or `staticmethod` on the main `PyEVMBackend` class?

```python
class PyEVMBackend:
    @staticmethod
     def generate_genesis_params(overrides=None):
        ...

    @staticmethod
    def generate_genesis_state(overrides=None):
        ...
```

Then the user workflow would be something like:

```python
from eth_tester import EthereumTester, PyEVMBackend
state = PyEVMBackend.generate_genesis_state(state_overrides)
genesis_params = PyEVMBackend.generate_genesis_params(genesis_overrides)
tester = EthereumTester(backend=PyEVMBackend(genesis_params, state)
```

it's a bit verbose, but not terrible and it should simplify the other APIs.

Reply:
:+1:  - I 100% agree.